src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -48,6 +59,9 @@ public class SocketOutput : ISocketOutput
             _connectionId = connectionId;
             _log = log;
             _tasksPending = new Queue<TaskCompletionSource<object>>();
+
+            _head = memory.Lease();
```

Comment:
might not need to lease a block here, if you go with `null` being nothing to write.